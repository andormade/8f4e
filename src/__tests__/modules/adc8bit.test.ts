import moduleTypeGenerator from '../../modules/adc8bit';

it('generates correct module type descriptor', () => {
	expect(moduleTypeGenerator()).toMatchSnapshot();
});
