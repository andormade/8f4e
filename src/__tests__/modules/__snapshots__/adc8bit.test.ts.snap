// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generates correct module type descriptor 1`] = `
Object {
  "buttons": Array [],
  "category": "Bitwise",
  "engine": Object {
    "config": Object {
      "resolution": 8,
    },
    "name": "adc",
  },
  "height": 252,
  "initialState": Object {},
  "inputs": Array [
    Object {
      "height": 13,
      "id": "in",
      "width": 12,
      "x": 12,
      "y": 42,
    },
  ],
  "lines": Array [
    Object {
      "height": 1,
      "width": 96,
      "x": 6,
      "y": 7,
    },
    Object {
      "height": 238,
      "width": 1,
      "x": 102,
      "y": 7,
    },
    Object {
      "height": 1,
      "width": 96,
      "x": 6,
      "y": 245,
    },
    Object {
      "height": 238,
      "width": 1,
      "x": 6,
      "y": 7,
    },
  ],
  "name": "8bit ADC",
  "outputs": Array [
    Object {
      "height": 13,
      "id": "out:1",
      "label": "bit0",
      "width": 12,
      "x": 84,
      "y": 42,
    },
    Object {
      "height": 13,
      "id": "out:2",
      "label": "bit1",
      "width": 12,
      "x": 84,
      "y": 56,
    },
    Object {
      "height": 13,
      "id": "out:3",
      "label": "bit2",
      "width": 12,
      "x": 84,
      "y": 70,
    },
    Object {
      "height": 13,
      "id": "out:4",
      "label": "bit3",
      "width": 12,
      "x": 84,
      "y": 84,
    },
    Object {
      "height": 13,
      "id": "out:5",
      "label": "bit4",
      "width": 12,
      "x": 84,
      "y": 98,
    },
    Object {
      "height": 13,
      "id": "out:6",
      "label": "bit5",
      "width": 12,
      "x": 84,
      "y": 112,
    },
    Object {
      "height": 13,
      "id": "out:7",
      "label": "bit6",
      "width": 12,
      "x": 84,
      "y": 126,
    },
    Object {
      "height": 13,
      "id": "out:8",
      "label": "bit7",
      "width": 12,
      "x": 84,
      "y": 140,
    },
  ],
  "sliders": Array [],
  "steppers": Array [],
  "width": 108,
}
`;
